internal/runners/initialize/init.go

```diff
@@ -47,22 +46,24 @@ func New(prime primeable) *Initialize {
 	return &Initialize{prime.Output(), prime.Project()}
 }
 
-func prepare(params *RunParams) error {
-	if params.Language == "" {
-		// Manually check for language requirement, because we need to fallback on the --language flag to support editor.V0
-		return locale.NewInputError("err_init_no_language", "You need to supply the [NOTICE]language[/RESET] argument, run `[ACTIONABLE]state init --help[/RESET]` for more information.")
-	}
-	langParts := strings.Split(params.Language, "@")
-	if len(langParts) > 1 {
-		params.version = langParts[1]
-	}
+func sanitize(params *RunParams, isheadless bool) error {
+	if !isheadless {
+		if params.Language == "" {
+			// Manually check for language requirement, because we need to fallback on the --language flag to support editor.V0
+			return locale.NewInputError("err_init_no_language", "You need to supply the [NOTICE]language[/RESET] argument, run `[ACTIONABLE]state init --help[/RESET]` for more information.")
+		}
+		langParts := strings.Split(params.Language, "@")
+		if len(langParts) > 1 {
+			params.version = langParts[1]
+		}
 
-	params.language = language.Supported{language.MakeByName(langParts[0])}
-	if !params.language.Recognized() {
-		return language.NewUnrecognizedLanguageError(
-			params.language.String(),
-			language.RecognizedSupportedsNames(),
-		)
+		params.language = language.Supported{language.MakeByName(langParts[0])}
+		if !params.language.Recognized() {
+			return language.NewUnrecognizedLanguageError(
+				params.language.String(),
+				language.RecognizedSupportedsNames(),
+			)
+		}
 	}
 
 	// Fail if target dir already has an activestate.yaml
```

Comment:
The block below would then need to be de-activated `if headless`

Reply:
`prepare` isn't called in `if headless`.